library/Vanilla/Addon.php

```diff
@@ -144,6 +144,12 @@ private function scanInfo() {
             if (empty($info)) {
                 throw new \Exception("The addon at $subdir has an empty info array.");
             }
+
+            $baseDir = explode('/', ltrim($subdir, '/'))[0];
+            if (in_array($baseDir, ['applications', 'plugins'])) {
```

Comment:
I don't want the new Addon class to have any knowledge of the directory it is in.